tests/qos/test_qos_sai.py

```diff
@@ -597,8 +615,25 @@ def testQosSaiHeadroomPoolSize(
         if not dutConfig['dualTor']:
             qosConfig['hdrm_pool_size']['pgs'] = qosConfig['hdrm_pool_size']['pgs'][:2]
             qosConfig['hdrm_pool_size']['dscps'] = qosConfig['hdrm_pool_size']['dscps'][:2]
+        src_dut_index = get_src_dst_asic_and_duts['src_dut_index']
+        dst_dut_index = get_src_dst_asic_and_duts['dst_dut_index']
+        src_asic_index = get_src_dst_asic_and_duts['src_asic_index']
+        dst_asic_index = get_src_dst_asic_and_duts['dst_asic_index']
+
+        # Need to adjust hdrm_pool_size src_port_ids, dst_port_id and pgs_num based on how many source and dst ports
+        # present
+        src_ports = dutConfig['testPortIds'][src_dut_index][src_asic_index]
+        if get_src_dst_asic_and_duts['src_asic'] == get_src_dst_asic_and_duts['dst_asic']:
+            # Src and dst are the same asics, leave one for dst port and the rest for src ports
+            qosConfig["hdrm_pool_size"]["src_port_ids"] = src_ports[:-1]
```

Comment:
This change causes failure on mellanox platform.
It expands port numbers of "qosConfig["hdrm_pool_size"]["src_port_ids"]".
But "qosConfig["hdrm_pool_size"].["pkts_num_trig_pfc_shp"] didn't expand accordingly.
Finally, caused "index out of range" error in ptf, as below:
```
"======================================================================",
"ERROR: sai_qos_tests.HdrmPoolSizeTest",
"----------------------------------------------------------------------",
"Traceback (most recent call last):",
"  File \"saitests/py3/sai_qos_tests.py\", line 2160, in runTest",
"    pkts_num_trig_pfc = self.pkts_num_trig_pfc_shp[i]",
"IndexError: list index out of range",
"",
"----------------------------------------------------------------------",
"Ran 1 test in 145.581s",
"",
"FAILED (errors=1)"]
```

For better understand error, share the my qos param for running mellanox qos test:

relevant qos param value before this change:
```
pkts_num_trig_pfc_shp=[50182, 25192, 12697, 6450, 3326, 1764, 983, 593, 398, 300, 251, 227, 215, 209]
src_port_ids=[2, 3, 4, 5, 6, 7, 8]
```

relevant qos parame value after apply this change:
```
pkts_num_trig_pfc_shp=[50182, 25192, 12697, 6450, 3326, 1764, 983, 593, 398, 300, 251, 227, 215, 209];
src_port_ids=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23];
```